host/client_integration_test.go

```diff
@@ -0,0 +1,372 @@
+	s.Assertions = require.New(s.T())
+}
+
+func (s *clientIntegrationSuite) TearDownTest() {
+
+}
+
+func (s *clientIntegrationSuite) setupSuite(enableGlobalDomain bool, isMasterCluster bool) {
+	// Have to define our overridden assertions in the test setup. If we did it earlier, s.T() will return nil
+	s.Assertions = require.New(s.T())
+	options := persistence.TestBaseOptions{}
+	options.ClusterHost = "127.0.0.1"
+	options.DropKeySpace = true
+	options.SchemaDir = ".."
+	options.EnableGlobalDomain = enableGlobalDomain
+	options.IsMasterCluster = isMasterCluster
+	s.SetupWorkflowStoreWithOptions(options, nil)
+
+	s.setupShards()
+
+	// TODO: Use mock messaging client until we support kafka setup onebox to write end-to-end integration test
+	s.mockProducer = &mocks.KafkaProducer{}
+	s.mockMessagingClient = mocks.NewMockMessagingClient(s.mockProducer, nil)
+
+	s.host = NewCadence(s.ClusterMetadata, s.mockMessagingClient, s.MetadataManager, s.ShardMgr, s.HistoryMgr, s.ExecutionMgrFactory, s.TaskMgr,
+		s.VisibilityMgr, testNumberOfHistoryShards, testNumberOfHistoryHosts, s.logger, 0, false)
+	s.host.Start()
+
+	s.engine = s.host.GetFrontendClient()
+	s.domainName = "integration-test-domain"
+	s.domainID = uuid.New()
+
+	s.MetadataManager.CreateDomain(&persistence.CreateDomainRequest{
+		Info: &persistence.DomainInfo{
+			ID:          s.domainID,
+			Name:        s.domainName,
+			Status:      persistence.DomainStatusRegistered,
+			Description: "Test domain for integration test",
+		},
+		Config: &persistence.DomainConfig{
+			Retention:  1,
+			EmitMetric: false,
+		},
+		ReplicationConfig: &persistence.DomainReplicationConfig{},
+	})
+}
+
+// testDataConverter implements encoded.DataConverter using gob
+type testDataConverter struct{}
+
+func (tdc *testDataConverter) ToData(value ...interface{}) ([]byte, error) {
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
+	for i, obj := range value {
+		if err := enc.Encode(obj); err != nil {
+			return nil, fmt.Errorf(
+				"unable to encode argument: %d, %v, with gob error: %v", i, reflect.TypeOf(obj), err)
+		}
+	}
+	return buf.Bytes(), nil
+}
+
+func (tdc *testDataConverter) FromData(input []byte, valuePtr ...interface{}) error {
+	dec := gob.NewDecoder(bytes.NewBuffer(input))
+	for i, obj := range valuePtr {
+		if err := dec.Decode(obj); err != nil {
+			return fmt.Errorf(
+				"unable to decode argument: %d, %v, with gob error: %v", i, reflect.TypeOf(obj), err)
+		}
+	}
+	return nil
+}
+
+func newTestDataConverter() encoded.DataConverter {
+	return &testDataConverter{}
+}
+
+func testActivity(ctx context.Context, msg string) (string, error) {
+	return "hello_" + msg, nil
+}
+
+var activityTaskList = "client-integration-data-converter-activity-tasklist"
+
+func testDataConverterWorkflow(ctx workflow.Context) (string, error) {
+	ao := workflow.ActivityOptions{
+		ScheduleToStartTimeout: time.Minute,
+		StartToCloseTimeout:    time.Minute,
+		HeartbeatTimeout:       20 * time.Second,
```

Comment:
remove HeartbeatTimeout as we dont need it here

Reply:
removed